internal/notifier/azure_eventhub.go

```diff
@@ -116,3 +135,68 @@ func newSASHub(address string) (*eventhub.Hub, error) {
 
 	return hub, nil
 }
+
+// newManagedIdentityToken is used to attempt credential-free authentication.
+func newManagedIdentityToken(ctx context.Context, proxy, serviceAccountName, providerName, providerNamespace string, tokenClient client.Client, tokenCache *pkgcache.TokenCache) (string, error) {
+	opts := []pkgauth.Option{pkgauth.WithScopes(azure.ScopeEventHubs)}
+	if proxy != "" {
+		proxyURL, err := url.Parse(proxy)
+		if err != nil {
+			return "", fmt.Errorf("error parsing proxy URL : %w", err)
+		}
+		opts = append(opts, pkgauth.WithProxyURL(*proxyURL))
+	}
+
+	if serviceAccountName != "" {
+		opts = append(opts, pkgauth.WithServiceAccount(types.NamespacedName{
+			Name:      serviceAccountName,
+			Namespace: providerNamespace,
+		}, tokenClient))
```

Comment:
Nit: for readability let's move the SA reference to a variable like `involvedObject` below:

```go
		serviceAccount := types.NamespacedName{
			Name:      serviceAccountName,
			Namespace: providerNamespace,
		}
		opts = append(opts, pkgauth.WithServiceAccount(serviceAccount, tokenClient))
```